### DISCUSSION TITLE
Transfère de bue entre 2 UC via une iframe

### DISCUSSION THREAD

**Discussion by MaxenceLe** [witsa/synapps#813] [General]
Bonjour, 

J'essai de transférer un vue de l'UC2 vers l'UC1 via une iframe.

Lorsque je créé la vue et que je rentre le nom de la scène, celui-ci m'indique "Scène non trouvé", alors que sur d'autre vue, j'ai procédé de la même manière et cela fonctionne très bien. De même manière en dupliquant l'iframe fonctionnant correctement et en changeant uniquement le nom de scène, cela m'indique toujours scène non trouvé, alors qu'elle existe belle et bien.

Ci-dessous les photos :
![image](https://github.com/user-attachments/assets/4a449c50-89af-4564-b965-d37ad71ffa83)
![image](https://github.com/user-attachments/assets/d81f875e-1c14-4085-95f9-ec456a68b0ae)

Je vous remercie,

**Comment by MaxenceLe** [comment]
La page de l'UC2 qui correspond *
![image](https://github.com/user-attachments/assets/77667d06-e4fc-4044-95af-fea0d4854db3)

**Comment by JeremieDENAEGEL** [comment]
Bonjour,

C'est un peu plus difficile de répondre sans la capture d'écran de ton acteur iframe pour voir où cela bloque.
Ceci étant, de base le synapps de l'UC1 ne peut afficher que ses scènes. Pour afficher un autre synapps, il faut actuellement passer par du wlog (?wlog=Utilisateur:MotDePasse).
Par contre, d'après les captures que tu as posté, quand tu vas appeller le synapps de l'UC2, cela ouvrira ta scène "Géographique Plateforme Vue Cellule 1".

Perso, je vois 2 solutions possibles mais il y en a peut-être d'autre
- Sur ton UC2 tu crées plusieurs "micro-synapps" correspondant aux scènes que tu veux afficher sur l'UC1.
Inconvérient : tu peux te retrouver avec beaucoup de synapps et si tu veux faire des modifications ultérieurement, il faudra repasser sur chaque synapps.
- Sur ton UC1 tu as une scène permettant de passer d'un automate à un autre et sur chaque automate tu as un menu ouvrant leurs sous-scènes.
Un peu dans ce principe :

![Exemple](https://github.com/user-attachments/assets/ba9a6226-b801-4665-b435-e0f1fd8084f3)


Pour ma part, j'utilise cette acteur pour ouvrir un synapps d'un autre automate voir du même automate en espérant qu'il t'aide ;)
```
SYNAPPS-STUDIO-ACTOR|{"type":"display/iframe","key":"iframe","properties":{"height":"870px","width":"500px","verticalAlignment":"top","allowTopNavigation":true},"additionalDefs":{"IP":{"type":"text","label":"REDY CIBLE: @ IP"},"Synapps":{"type":"text","label":"REDY CIBLE: Label Synapps"},"wlog":{"type":"text","label":"REDY CIBLE: wlog"}},"events":{"onPostInit":["this.properties.url = this.additionals.IP + '/easy/Synapps/' + this.additionals.Synapps + '/?wlog=' + this.additionals.wlog"]}}
```
